src/router_core/connections.c

```diff
@@ -1702,16 +1702,40 @@ static void qdr_link_inbound_first_attach_CT(qdr_core_t *core, qdr_action_t *act
     qdr_connection_t      *conn = safe_deref_qdr_connection_t(action->args.connection.conn);
     qdr_link_t            *link = safe_deref_qdr_link_t(action->args.connection.link);
     qdr_delivery_t *initial_dlv = action->args.connection.initial_delivery;
+
+    qdr_terminus_t *source = action->args.connection.source;
+    qdr_terminus_t *target = action->args.connection.target;
+
     if (discard || !conn || !link) {
         if (initial_dlv)
             qdr_delivery_decref(core, initial_dlv,
                                 "qdr_link_inbound_first_attach_CT - discarding action");
+
+        if (discard) {
```

Comment:
Remove this conditional - you want to free these terminus if !conn or !link as well since in all those cases the source/target are leaked.